vhdl_lang/src/config.rs

```diff
@@ -88,7 +90,58 @@ impl LibraryConfig {
             }
         }
         // Remove duplicate file names from the result
-        result.into_iter().unique().collect()
+        result = result.into_iter().unique().collect();
```

Comment:
I think that, if you are already collecting everything into an iterator, maybe using  a `BTreeSet` or a `HashSet` from the beginning might a better alternative? With the new approach, there is a lot of juggling with vectors and other iterables which adds unwanted complexity.

Reply:
I've implemented this with `BTreeSet`, but the test results are no non-deterministic. `config::tests::config_from_str` in particular seems to fail approximately half of the time due to the ordering of the files in the `assert!` on `&[pkg2_path, absolute_vhd]`. The same thing occurs with `HashSet`.

How important is the returned file order for the rest of the VHDL-LS? If it isn't, then I could change the tests to check the contents of the collection without the order, but only if it really doesn't matter.

Reply:
I don't think the file order plays any role. `vhdl_ls` should is capable of determining the compile order independent on the actual order of files that were passed in, so you can just re-write your test to compare the expected with the actual disregarding the order.

Reply:
Thanks. I've done that and all seems to be good now.